docs/modal.md

```diff
@@ -25,48 +25,104 @@ use Yiisoft\Yii\Bulma\Asset\BulmaJsAsset;
+        rootEl.classList.add('is-clipped');
+        $target.classList.add('is-active');
+    }
+
+    function closeModals() {
+        rootEl.classList.remove('is-clipped');
+        $modals.forEach(function ($el) {
+            $el.classList.remove('is-active');
+        });
+    }
+
+    function getAll(selector) {
+        var parent = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : document;
+
+        return Array.prototype.slice.call(parent.querySelectorAll(selector), 0);
+    }
+
+    document.addEventListener('keydown', function (event) {
+        var e = event || window.event;
+
+        if (e.keyCode === 27) {
+            closeModals();
+            closeDropdowns();
+        }
+    });
+JS;
+
+$this->registerJs($modalJS);
+?>
+
+<?= Modal::widget()->id('modal')->begin() .
+        Div::tag()->class('box')->content('Say hello...')->render() . PHP_EOL .
+    Modal::end() ?>
 ```
 
-HTML produced is like the following:
+The code above generates the following HTML:
+
 ```html
-<button type="button" class="button" data-target="#w1-modal" aria-haspopup="true">Toggle button</button>
-<div id="w1-modal" class="modal">
+<button class="button modal-button" data-target="#modal" aria-haspopup="true">Toggle button</button>
+<div id="modal" class="modal">
     <div class="modal-background"></div>
+    <button class="modal-close" aria-label="close"></button>
     <div class="modal-content">
-        <div class="box">
-            Say hello...
-        </div>
+        <div class="box">Say hello...</div>
     </div>
-    <button type="button" class="modal-close is-large" aria-label="close"></button>
 </div>
 ```
 
-## Reference
+## Setters
+
+All setters are immutable and return a new instance of the `Yiisoft\Yii\Bulma\Modal` class with the specified value.
 
 Method | Description | Default
 -------|-------------|---------
-`id(string $value)` | Widget ID. | `''`
+`attributes(array $value)` | The HTML attributes. | `[]`
 `autoIdPrefix(string $value)` | Prefix to the automatically generated widget ID. | `w`
-`options(array $value)` | HTML attributes for the widget container tag. | [`class` => `modal`]
-`contentOptions(array $value)`| HTML attributes for the widget content tag. | `[]`
-`closeButtonOptions(array $value)` | HTML attributes for the widget button tag. | [`class` => `modal-close`, `aria-label` => `close`]
-`closeButtonSize(string $value)` | Size close button. | `is-small`, `is-medium`, `is-large`
+`closeButtonAttributes(array $value)` | HTML attributes for the button tag. | [`class` => `modal-close`, `aria-label` => `close`]
+`closeButtonSize(string $value)` | Size close button. Options available are: (`Modal::SIZE_SMALL`, `Modal::SIZE_MEDIUM`, `Modal::SIZE_LARGE`). | Default setting empty normal.
+`contentAttributes(array $value)`| HTML attributes for the content tag. | `[]`
+`id(string $value)` | Widget ID. | `''`
+`modalBackgroundClass(string $value)` | Class for the modal background. | `modal-background`
```

Comment:
I don't think we need to mention "modal".

Reply:
done.